docs/connect.rst

```diff
@@ -0,0 +1,245 @@
+.. _connect:
+
+==================
+Connect to CrateDB
+==================
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+.. _data-source-name:
+
+Data Source Names
+=================
+
+PDO makes use of `Data Source Name`_ (DSN) strings.
+
+A basic version of the DSN string for CrateDB might be::
+
+    crate:<HOST_ADDR>:<PORT>
+
+Here, replace ``<HOST_ADDR>`` with the host address of a CrateDB node, and
+replace ``<PORT>`` with the port number of the  `HTTP endpoint`_.
+
+Example DSN strings:
+
+- ``crate:localhost:4200``
+- ``crate:crate-1.vm.example.com:4200``
+- ``crate:198.51.100.1:4200``
+
+You can specify a second CrateDB node, like so::
+
+    crate:<HOST_ADDR_1>:<PORT>,<HOST_ADDR_2>:<PORT>
+
+Here, ``<HOST_ADDR_1>`` and ``<HOST_ADDR_2>`` are the host strings for the
+first and second CrateDB nodes, respectively.
+
+In fact, you can specify as many nodes as you like. Each corresponding host
+string must be separated from the previous one using a ``,`` character.
+
+The driver will attempt to connect to each node in the order they appear. The
+first successul connection will be used, and all other nodes will be ignored
+for the duration of that connection.
+
+You can also specify a schema, like so::
+
+    crate:<HOST_ADDR>:<PORT>/<SCHEMA>
+
+Here, replace ``<SCHEMA>`` with the name of the schema you want to select.
+
+.. TIP::
+
+   The default CrateDB schema is ``doc``, and if you do not specify a schema,
+   this is what will be used.
+
+   However, you can query any schema you like at query time.
+
+So, to wrap up, here's a more complex list of example DSN strings:
+
+- ``crate:localhost:4200/my_schema``
+- ``crate:crate-1.vm.example.com:4200,crate-2.vm.example.com:4200``
+- ``crate:198.51.100.1:4200,198.51.100.2:4200/another_schema``
+
+.. _get-connection:
+
+Get a Connection
+================
+
+You can get a PDO connection like this:
+
+.. code-block:: php
+
+   use Crate\PDO\PDO as PDO;
+
+   $dsn = '<DATA_SOURCE_NAME>';
+   $user = 'crate';
+   $password = null;
+   $options = null;
+   $connection = new PDO($dsn, $user, $password, $options);
+
+.. NOTE::
+
+   Authentication was introduced in CrateDB versions 2.1.x.
+
+   If you are using CrateDB 2.1.x or later, you must supply a username. If you
+   are using earlier versions of CrateDB, this argument is not supported.
+
+   See the :ref:`compatibility notes <cratedb-versions>` for more information.
+
+.. TIP::
+
+   If you have not configured a custom `database user`_, you probably want to
+   authenticate as the CrateDB superuser, which is ``crate``. By default, the
```

Comment:
It's not possible to define a password for the `crate` superuser, so it is not by default.